src/components/SingleArticle.jsx

```diff
@@ -2,22 +2,40 @@ import React, { useEffect, useState } from "react";
 import { useParams } from "react-router-dom";
 import Header from "./Header";
 import Comments from "./Comments";
-import { fetchSingleArticle } from "../utils/api";
+import { addVotes, fetchSingleArticle } from "../utils/api";
 import "../css/singlearticle.css";
 import "../css/comments.css";
 
-const SingleArticle = () => {
+const SingleArticle = ({ setVotes }) => {
   const { article_id } = useParams();
   const [article, setArticle] = useState({});
+  const [newVote, setNewVote] = useState(0);
+  const [hasVoted, setHasVoted] = useState(false);
   const [isLoading, setIsLoading] = useState(true);
 
   useEffect(() => {
     fetchSingleArticle(article_id).then((response) => {
       setArticle(response);
+      setNewVote(response.votes); //set votes from article
```

Comment:
Setting multiple states like this is generally considered bad practice because it creates duplicate sources of truth. Here’s why this can lead to issues such as a lack of synchronisation; If the article changes (e.g., the article data is updated), the votes state won't automatically update to reflect this change. This can lead to stale or inconsistent data in the UI.

better to have a updatedVotes state and derive the votes in when rendering your article with  article.votes + updatedVotes

Reply:
It probably wont matter in this case but could cause issues in other parts of your code if you implement this patten